### ISSUE TITLE
Limit the amount of grab pan area beyond the image

### ISSUE THREAD

**Issue by p-4** [kingdido999/zooming#62]
Is there a way to limit the grab pan area?

Possibly with a margin option with the ability to specify the amount in px.
Then the maximum amount of white color one could see from the edge of the image to the edge of the browser would be this margin amount in px.

I have very limited coding experience and do not know how to approach this.
Any help greatly appreciated.

![grab_pan_limit_margin](https://user-images.githubusercontent.com/37654737/37761593-64b1f306-2dba-11e8-9e06-f7956d0478df.PNG)

**Comment by kingdido999** [comment]
If I understand your intention correctly, you wanted to keep the image body within a specified frame while grabbing to avoid part of image goes out of the browser window. There is no option to achieve this right now.

We also need to consider the potential UX issue it brings up if we add this option. For example, if an image is already scaled to fit the window, does it respect this option? Would we allow it to even scale further when we grab it? Also, it might look a little bit odd since at some point the image will just stop moving according to our grab intention...

**Comment by p-4** [comment]
> you wanted to keep the image body within a specified frame while grabbing to avoid part of image goes out of the browser window.

Not really.  The idea is to avoid seeing too much white space, and limiting the amount of pixels the image can travel from the edge of the browser. So this would not conflict with other features I think.

I believe it is better to limit this, as the viewer does not need to be able to move the image so much that it goes half outside of his viewable area (just like I did on the attached screenshot).

This would improve the user experience even more when the background has transparency and the webpage is visible behind the zoomed image.

On the screenshot I have also put red lines to represent how much I can pan (grab) the image and see a lot of white space. The idea would be to be able to limit this and set a smaller distance (the blue lines).

![capture2](https://user-images.githubusercontent.com/37654737/37898084-2b5292d8-30e8-11e8-8d20-c1ed95817d4b.png)

**Comment by kingdido999** [comment]
Ok I got your point now! I agree such option will improve the user experience.
